From the POV of YAMCS:

Directory requests do not work because upstream issue.
Put requests work and end in `AttributeError: 'NoneType' object has no attribute 'source_file'`
Proxy put requests work and end in `AssertionError: assert self._params.fp.file_size is not None`

--- a/oresat_c3/services/edl.py
+++ b/oresat_c3/services/edl.py
@@ -695,7 +695,8 @@ class EdlFileReciever(CfdpUserBase):
                 elif isinstance(response, PutRequest):
                     put = response
             # Ignore non-reserved messages for now
-        self.scheduled_requests.put(put)
+        if put is not None:
+            self.scheduled_requests.put(put)
 
     def file_segment_recv_indication(self, params: FileSegmentRecvdParams):
         logger.info(f"Indication: File Segment Recv. {params}")

